File: examples/app-yaml-config/sdk/mjs/06-resolver-env.mjs
#!/usr/bin/env node
// F03 / story 02 — {{env.X}} + default-value syntax.
import { createResolver, MissingStrategy } from "@ployglot/runtime-template-resolver";

const resolver = createResolver({ missingStrategy: MissingStrategy.IGNORE });

const out = {
  home_set: !!process.env.HOME,
  default_when_unset: await resolver.resolve('{{env.NEVER_SET_BY_TEST | "default-here"}}', { env: process.env }),
  default_numeric: await resolver.resolve('{{env.ALSO_NEVER_SET | "42"}}', { env: process.env }),
};
// Don't print actual $HOME (host-specific) — just confirm it resolves.
out.home_resolved = (await resolver.resolve("{{env.HOME}}", { env: process.env })) ? "<HOME>" : null;

console.log(JSON.stringify(out, null, 2));

File: examples/app-yaml-config/sdk/py/06-resolver-env.py
#!/usr/bin/env python3
"""F03 / story 02 — {{env.X}} + default-value syntax."""
from __future__ import annotations

import asyncio
import json
import os

from runtime_template_resolver import MissingStrategy, create_resolver


async def main():
    r = create_resolver(missing_strategy=MissingStrategy.IGNORE)
    out = {
        "home_set": "HOME" in os.environ,
        "default_when_unset": await r.resolve('{{env.NEVER_SET_BY_TEST | "default-here"}}', {"env": dict(os.environ)}),
        "default_numeric": await r.resolve('{{env.ALSO_NEVER_SET | "42"}}', {"env": dict(os.environ)}),
        "home_resolved": "<HOME>" if await r.resolve("{{env.HOME}}", {"env": dict(os.environ)}) else None,
    }
    print(json.dumps(out, indent=2))


if __name__ == "__main__":
    asyncio.run(main())
